### PR TITLE
[FIXED] JetStream: data race between consumer and stream when updated

### PR DESCRIPTION
**Pull request by kozlovic**
Some tests were showing often a race between updateWithAdvisory() and isFiltered().

There may be other data races in other files that read the mset's configuration without proper locking, but this PR addresses specifically issues between consumer (consumer.go) and stream.

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: server/consumer.go
Comment (kozlovic) [inline server/consumer.go:720]: `meet.getCfg()` may not even be needed here, since meet.setCfg() is still called under mset.mu.Lock().

File: server/stream.go
Comment (kozlovic) [inline server/stream.go:2051]: If it is agreed that call to `getCfg()` in `addConsumerWithAssignment()` is not needed, then this method may not be needed at all (for now at least).

**Review by kozlovic — COMMENTED** [review]
@derekcollison I think I don't even need to call to mset.getCfg() since we are protected by the stream's lock there. If you agree, I will update the PR tomorrow.

File: server/consumer.go
Comment (kozlovic) [reply server/consumer.go]: I have reverted that part in the new commit.

File: server/stream.go
Comment (kozlovic) [reply server/stream.go]: I have removed it, and setCfg() too (I did inline it since there was only one place we were using it).

Comment (derekcollison) [reply server/stream.go]: I don't think it is needed. With it not there can you see the race?

**Review by derekcollison — APPROVED** [review]
LGTM

**Comment by derekcollison** [comment]
I will merge after all green.

**Comment by kozlovic** [comment]
@derekcollison I got a data race on a test for completely different thing (non JS). Can I add it here or do you prefer new PR for that?

**Comment by kozlovic** [comment]
Sorry, I did not wait for your answer. It was a quick fix, so added to this PR.

**Comment by kozlovic** [comment]
@derekcollison Ok, got all green, ready to merge!

**Pull request merged**
